### PR TITLE
Compile x265 with 10/12 bits support.

### PR DESCRIPTION
**Pull request by corco**
This should add support for 10 bits and 12 bits encodes for x265. Right now, the following code fails for lack of 10 bits support (note the code works if you replace "libx265" for "av1")

Note that according to x265's [CMakeLists.txt](https://github.com/videolan/x265/blob/419182243fb2e2dfbe91dfc45a51778cf704f849/source/CMakeLists.txt#L387), the option is only valid for 64 bits architectures.
```
import av

icntnr = av.open("img.png")

ivstrm = next(s for s in icntnr.streams if s.type == "video")

ocntnr = av.open("test.mp4", "w")
ovstrm = ocntnr.add_stream("libx265")
ovstrm.width = ivstrm.width
ovstrm.height = ivstrm.height
ovstrm.pix_fmt = "gbrp10"

for packet in icntnr.demux(icntnr.streams.video[0]):
    if packet.stream.type != "video":
        continue

    for ifrm in packet.decode():
        print(ifrm)

        for p in ovstrm.encode(ifrm):
            ocntnr.mux(p)

for packet in ovstrm.encode():
    ocntnr.mux(packet)

ocntnr.close()
```

### CODE REVIEW

**Comment by jlaine** [comment]
Hi! I believe the solution is quite a bit more involved than this. From what I've seen in some repos it involves compiling x265 in several iterations each time for a specific bit depth. Have you checked the resulting library locally?

**Comment by corco** [comment]
Yes, you are right.

I was mislead by the fact that the resulting library accept 8, 10 and 12 bits pixel formats, but on closer inspection, it encodes all of them as 12 bits.

I will try to build it like the debian package.

**Comment by corco** [comment]
I changed the PR to build much like the debian package.

I tested on linux x86_64, this time the output has the proper bit depth.

File: scripts/cibuildpkg.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import tarfile
 import tempfile
 import time
-from dataclasses import dataclass, field
+from dataclasses import dataclass, field, replace
```
Comment (jlaine) [inline scripts/cibuildpkg.py:13]: Well, I learned something today, I did not know `replace`!

**Comment by jlaine** [comment]
Thanks for this! Let's see how CI goes on the non-Linux platforms. I see an unrelated failure on macos (some mirrors are flakey), I'll restart it if it's the only failure.

**Comment by jlaine** [comment]
Wow, major kudos here, I'm stunned that the build went through on all Linux + Windows builds!

I'm re-triggering the macOS builds as their failures were not related to x265.

**Comment by corco** [comment]
> Wow, major kudos here, I'm stunned that the build went through on all Linux + Windows builds!
> 
> I'm re-triggering the macOS builds as their failures were not related to x265.

Well, to be fair, I did run the workflow on my private repo until it pass!

**Comment by jlaine** [comment]
> Well, to be fair, I did run the workflow on my private repo until it pass!

I do love that feature of GitHub Actions, it's a shame not everyone is aware of it :)

Merged, thanks for your contribution! I will use the new builds for the future PyAV releases.

**Pull request merged**